Apply changes for the upcoming build infrastructure improvements (2.x)

warpech opened the pull request:
point 8 assumes changes from: https://github.com/Starcounter/AdminTrack/issues/438

point 4 assumes changes from: https://github.com/Starcounter/AdminTrack/issues/435

@Starcounter/webplatformteam, @ubbeK, @joozek78 please review.

warpech committed:
apply changes for the upcoming build infrastructure improvements

point 8 assumes changes from: https://github.com/Starcounter/AdminTrack/issues/438

point 4 assumes changes from: https://github.com/Starcounter/AdminTrack/issues/435

ubbeK commented on CONTRIBUTING.md:
```diff
@@ -1,4 +1,4 @@
-This repo is supposed to replace [/Starcounter/level1/tree/develop/src/BuildSystem/ClientFiles](https://github.com/Starcounter/level1/tree/develop/src/BuildSystem/ClientFiles). At this point it does not replace it yet, but it is the source of truth for it.
+This repo is the source of client files for [/Starcounter/level1/tree/develop/src/BuildSystem/ClientFiles](https://github.com/Starcounter/level1/tree/develop/src/BuildSystem/ClientFiles).
```
The link is wrong, this should not exist anymore (I can see that not all ClientFiles are removed though, will create an issue for that: https://github.com/Starcounter/level1/tree/develop/src/BuildSystem/ClientFiles)

I think you should say something like, source of client files found in `%StarcounterBin%/ClientFiles`

ubbeK replied:
Issue has been created to remove remaining ClientFiles from Level1: https://github.com/Starcounter/level1/issues/4485

warpech replied:
Thanks, fixed.

warpech committed:
problem: the link to client files in level1 is no longer correct

solution: change the link to information about what is the folder on disk

tomalec reviewed (approved):
LGTM

warpech committed:
clarifications

move basic information about the repo and branching to README.md
clarifly the developer process in CONTRIBUTING.md

warpech commented:
@ubbeK I made a few clarifications, especially changed the last point to be:

> 8. Merge the PR to StarcounterClientFiles branch `2.x`.
>   - at this point, a new build of Level1 is started

Is this correct? Does Level1 have a trigger on that?

warpech committed:
clarify about how does pushing to 2.x affect the Level1 triggers
